### DISCUSSION TITLE
Copilot Chat Agent Extensibility

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#923] [Announcements]
All the details in this blog post under the extensibility section:
https://code.visualstudio.com/blogs/2023/11/13/vscode-copilot-smarter#_extensibility

Let us know what you think and I am happy to answer any questions. Thank you!

**Comment by Wladefant** [comment]
Very good. I am already creating these agents and am mindblown by the api. Could you tell me about the timeline a bit? And there are multiple proposed apis. Are they all going to be out together or not?

**Reply by isidorn** [reply]
Glad to hear you like the api.
We plan to finalize the chatAgent2 in the near future. Once we have the exact month set I will comment here.

There are other APIs that are in proposal, and we are still working on them so it is early to talk about when they will be in the final state.

Hope that helps. If you have more questions do let me know.

**Reply by Wladefant** [reply]
What timeline are we talking here about: months or ca. one month. And you mean the proposed apis chat, chatagent2 + additions, chatrequestaccess?

**Reply by isidorn** [reply]
Definitely more than one month. I think less than 6 months if I would have to guess now for chatagent2. For chatrequestaccess we are still in discussions, once we have more details I will share here.

**Comment by isidorn** [comment]
Closing in favor of https://github.com/microsoft/vscode-discussions/discussions/1210
